### ISSUE TITLE
setDF's rownames argument could be a column to be converted to row names

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#1719]
Say we had

    DT <- data.table(a = c("a", "b", "c"), b = 1:3)

And we wanted

    data.frame(b = 1:3, row.names = c("a", "b", "c"))

It seems we could do this all by reference via:

    rn <- DT$a
    DT[ , a := NULL]
    setDF(DT, rownames = rn)

Seems natural to implement this via

    setDF(DT, rownames = "a")

Or else add an argument:

   setDF(DT, to.row.names = "a")

**Comment by mbacou** [comment]
@MichaelChirico in case your question is related to using data.table in combination with spatial classes (for which rownames matter), like SpatialPolygonsDataFrame or SpatialPointsDataFrame etc..  then you can already use the argument `match.ID`:

```
SpatialPolygonsDataFrame(polygons, setDF(DT), match.ID="rn")
SpatialPolygonsDataFrame(points, setDF(DT), match.ID="rn")
```

**Comment by arunsrinivasan** [comment]
@MichaelChirico I think this functionality is much more reasonable than the current one (of providing a vector). Perhaps best to replace the current functionality with this one? I don't think many people use the `rownames` argument.. shouldn't be a problem to just change it... 

Do you've time for a PR? :-)

**Comment by MichaelChirico** [comment]
of course! maybe later this weekend

**Comment by MichaelChirico** [comment]
@arunsrinivasan should I add some sort of deprecation message for the old usage?

**Comment by MichaelChirico** [comment]
@arunsrinivasan also, we have the `is.data.frame(x)` branch to deal with. Current logic is:

```
if (is.data.frame(x)) {
  if (!is.null(rownames)){
    if (length(rownames) != nrow(x)) 
      stop("rownames incorrect length; expected ", nrow(x), " names, got ", length(rownames))
    setattr(x, "row.names", rownames)
  }
  x
}
```

Let's just drop this whole branch?

Otherwise, we'll have to `setDT(x)` in order to be able to drop the named column by reference (if I'm not mistaken)...

**Comment by arunsrinivasan** [comment]
Makes sense. `rownames` argument could be valid just when input is `data.table`.. 

If we deprecate, then just the warning should be in place for this release. But I think this feature affects very less people (if at all), that we could directly change it.
